src/main/java/com/synopsys/integration/alert/provider/blackduck/new_collector/PolicyViolationMessageBuilder.java

```diff
@@ -0,0 +1,456 @@
+    public static final String CATEGORY_TYPE = "Policy";
+    public static final String VULNERABILITY_CHECK_TEXT = "vuln";
+    private final Logger logger = LoggerFactory.getLogger(PolicyViolationMessageBuilder.class);
+    private final Map<String, ComponentItemPriority> policyPriorityMap = new HashMap<>();
+    private VulnerabilityUtil vulnerabilityUtil;
+
+    @Autowired
+    public PolicyViolationMessageBuilder(VulnerabilityUtil vulnerabilityUtil) {
+        this.vulnerabilityUtil = vulnerabilityUtil;
+        policyPriorityMap.put("blocker", ComponentItemPriority.HIGHEST);
+        policyPriorityMap.put("critical", ComponentItemPriority.HIGH);
+        policyPriorityMap.put("major", ComponentItemPriority.MEDIUM);
+        policyPriorityMap.put("minor", ComponentItemPriority.LOW);
+        policyPriorityMap.put("trivial", ComponentItemPriority.LOWEST);
+        policyPriorityMap.put("unspecified", ComponentItemPriority.NONE);
+    }
+
+    @Override
+    public String getNotificationType() {
+        return NotificationType.RULE_VIOLATION.name();
+    }
+
+    @Override
+    public List<ProviderMessageContent> buildMessageContents(final Long notificationId, final Date providerCreationDate, final ConfigurationJobModel job, final RuleViolationNotificationView notificationView,
+        final BlackDuckBucket blackDuckBucket,
+        final BlackDuckServicesFactory blackDuckServicesFactory) {
+        long timeout = blackDuckServicesFactory.getBlackDuckHttpClient().getTimeoutInSeconds();
+        BlackDuckBucketService bucketService = blackDuckServicesFactory.createBlackDuckBucketService();
+        BlackDuckResponseCache responseCache = new BlackDuckResponseCache(bucketService, blackDuckBucket, timeout);
+        BlackDuckService blackDuckService = blackDuckServicesFactory.createBlackDuckService();
+        ComponentService componentService = blackDuckServicesFactory.createComponentService();
+        RuleViolationNotificationContent violationContent = notificationView.getContent();
+        ItemOperation operation = ItemOperation.ADD;
+        try {
+            ProviderMessageContent.Builder projectVersionMessageBuilder = new ProviderMessageContent.Builder()
+                                                                              .applyProvider(getProviderName(), blackDuckServicesFactory.getBlackDuckHttpClient().getBaseUrl())
+                                                                              .applyTopic("Project", violationContent.getProjectName())
+                                                                              .applySubTopic("Project Version", violationContent.getProjectVersionName(), violationContent.getProjectVersion())
+                                                                              .applyAction(operation)
+                                                                              .applyNotificationId(notificationId)
```

Comment:
We don't need to apply the action and notification id at the top level if there are component items. It won't have any affect in this case, but if there's ever a case where no component items are present, channels will treat it like a PROJECT or PROJECT_VERSION notification.